src/Data/Bin/Properties.agda

```diff
@@ -1,143 +1,792 @@
-1<[23] {b} = less (ℕₚ.m≤n+m 2 (Fin.toℕ b))
-
-1<2+ : ∀ {b bs} → [] 1# < (b ∷ bs) 1#
-1<2+ {_} {[]}     = 1<[23]
-1<2+ {_} {b ∷ bs} = <-trans 1<[23] (∷ʳ-mono-< {a = b} 1<2+)
-
-0<1+ : ∀ {bs} → 0# < bs 1#
-0<1+ {[]}     = less (s≤s z≤n)
-0<1+ {b ∷ bs} = <-trans (less (s≤s z≤n)) 1<2+
-
-<⇒≢ : ∀ {a b} → a < b → a ≢ b
-<⇒≢ lt eq = asym⟶irr (resp₂ _<_) sym <-asym eq lt
-
-<-cmp : Trichotomous _≡_ _<_
-<-cmp 0#            0#            = tri≈ (<-irrefl refl) refl (<-irrefl refl)
-<-cmp 0#            (_ 1#)        = tri< 0<1+ (<⇒≢ 0<1+) (<-asym 0<1+)
-<-cmp (_ 1#)        0#            = tri> (<-asym 0<1+) (<⇒≢ 0<1+ ∘ sym) 0<1+
-<-cmp ([] 1#)       ([] 1#)       = tri≈ (<-irrefl refl) refl (<-irrefl refl)
-<-cmp ([] 1#)       ((b ∷ bs) 1#) = tri< 1<2+ (<⇒≢ 1<2+) (<-asym 1<2+)
-<-cmp ((a ∷ as) 1#) ([] 1#)       = tri> (<-asym 1<2+) (<⇒≢ 1<2+ ∘ sym) 1<2+
-<-cmp ((a ∷ as) 1#) ((b ∷ bs) 1#) with <-cmp (as 1#) (bs 1#)
-... | tri<  lt ¬eq ¬gt =
-  tri< (∷ʳ-mono-< lt)  (<⇒≢ (∷ʳ-mono-< lt)) (<-asym (∷ʳ-mono-< lt))
-... | tri> ¬lt ¬eq  gt =
-  tri> (<-asym (∷ʳ-mono-< gt)) (<⇒≢ (∷ʳ-mono-< gt) ∘ sym) (∷ʳ-mono-< gt)
-... | tri≈ ¬lt refl ¬gt with 𝔽ₚ.<-cmp a b
-...   | tri≈ ¬lt′ refl ¬gt′ =
-  tri≈ (<-irrefl refl) refl (<-irrefl refl)
-...   | tri<  lt′ ¬eq  ¬gt′ =
-  tri< (∷ˡ-mono-< lt′)  (<⇒≢ (∷ˡ-mono-< lt′)) (<-asym (∷ˡ-mono-< lt′))
-...   | tri> ¬lt′ ¬eq  gt′  =
-  tri> (<-asym (∷ˡ-mono-< gt′)) (<⇒≢ (∷ˡ-mono-< gt′) ∘ sym) (∷ˡ-mono-< gt′)
-
-<-isStrictTotalOrder : IsStrictTotalOrder _≡_ _<_
-<-isStrictTotalOrder = record
+≡-decSetoid : DecSetoid 0ℓ 0ℓ
+≡-decSetoid = record
+  {isDecEquivalence = ≡-isDecEquivalence
+  }
+
+2suc≢0 : ∀ {x} → 2suc x ≢ 0#
+2suc≢0 ()
+
+suc2*≢0 : ∀ {x} → suc2* x ≢ 0#
+suc2*≢0 ()
+
+------------------------------------------------------------------------
+-- Properties of 2*
+------------------------------------------------------------------------
+
+2suc-as∘ : 2suc ≗ 2* ∘ suc
+2suc-as∘ 0#        =  refl
+2suc-as∘ (2suc x)  =  cong 2suc (2suc-as∘ x)
+2suc-as∘ (suc2* x) =  refl
+
+suc2*-as∘ : suc2* ≗ suc ∘ 2*
+suc2*-as∘ 0#        = refl
+suc2*-as∘ (2suc x)  = refl
+suc2*-as∘ (suc2* x) = begin
+  suc2* (suc2* x)     ≡⟨ cong suc2* (suc2*-as∘ x) ⟩
+  suc2* (suc 2x)      ≡⟨⟩
+  suc (2suc 2x)       ≡⟨ cong suc (2suc-as∘ 2x) ⟩
+  suc (2* (suc 2x))   ≡⟨ cong (suc ∘ 2*) (sym (suc2*-as∘ x)) ⟩
+  suc (2* (suc2* x))  ∎
+  where 2x = 2* x
+
+2x≡0⇒x≡0 : ∀ {x} → 2* x ≡ 0# → x ≡ 0#
+2x≡0⇒x≡0 {0#} 2x≡0 = refl
+
+x≢0⇒2x≢0 : ∀ {x} → x ≢ 0# → 2* x ≢ 0#
+x≢0⇒2x≢0 x≢0 = x≢0 ∘ 2x≡0⇒x≡0
+
+2*≢1 : ∀ {x} → 2* x ≢ 1B
+2*≢1 {0#} ()
+
+------------------------------------------------------------------------
+-- Properties of suc/pred
+------------------------------------------------------------------------
+
+suc≢0 : ∀ {x} → suc x ≢ 0#
+suc≢0 {0#}      ()
+suc≢0 {2suc _}  ()
+suc≢0 {suc2* _} ()
+
+pred∘suc : pred ∘ suc ≗ id
+pred∘suc 0#        =  refl
+pred∘suc (2suc x)  =  sym (2suc-as∘ x)
+pred∘suc (suc2* x) =  refl
+
+suc∘pred : ∀ x → x ≢ 0# → suc (pred x) ≡ x
```

Comment:
We tend to use the `-` symbol instead of `∘` in names in the standard library. The reason for this is that `∘` can't always be used in the type itself, as this example above demonstrates. Could you replace it in all the names please?

Reply:
Done